assembler/assemble.py

```diff
@@ -825,6 +825,8 @@ def args_ry(self, tokens,opcode):
     def args_rgm(self, tokens,opcode):
         #BIT BSET BCLR BTG
         arg_count_test(len(tokens),3)
+        if tokens[1] in ["RS", "IN", "OUT"]:
```

Comment:
The official docs don't mention IN or OUT being usable as an alias for RG, so I would keep just RS for consistency.

Reply:
![image](https://user-images.githubusercontent.com/166042/204321141-aa48c76c-9077-42a8-8d1b-9571df4ab2de.png)

This is where I got `if G=3, then it's actually using IN` from. This is from the `BIT` documentation. 

Similarly, for BTG/BSET/BCLR, it says the same, and actually still says `IN` for `G=3`, but that's a mistake, as the diagram that accompanies it shows it's using the `OUT` register (and using `IN` for BSET/BCLR/BTG would be nonsensical).

![image](https://user-images.githubusercontent.com/166042/204322029-fa746012-f6be-4e04-b1b7-6404e5354da3.png)

Reply:
Oh yeah, I'm not contesting that the instruction _operates_ on `IN` or `OUT` registers, but the syntax that's documented only supports `RS`:

![Screenshot_20221128-134513](https://user-images.githubusercontent.com/1809626/204367434-1150a17f-105c-4c2a-a282-cb8ebfbb71e5.png)

That's why I don't think it's worth supporting additional aliases as it adds undocumented features that make alternate assembler implementations harder. The documentation doesn't even support `R3`, but at this point it should be still supported for compatibility with existing code.

Reply:
It's all a bit wishy-washy, but I contend that's the purpose of aliases. As far as R3 not being supported, it's actually the example given above in the BCLR example :laughing: 

![image](https://user-images.githubusercontent.com/166042/204369689-97ee7eb4-67d8-42a6-b2bf-c3d31faff74e.png)

If the instruction operates on `IN/OUT`, why *not* support using aliases that make that clearer in the assembler?

Reply:
Good point on R3.

Here's another aspect. For MOV and other instructions OUT/IN are aliases specifically for R10/R11 since the assembler can't know the status of WrFlags at runtime and there's no special value to redirect to the active pair for those instructions. Relocating the registers doesn't change that OUT always resolves to R10.

In contrast, RS is a special notation that points to one of OUT/IN or OUTB/INB, which is resolved at runtime.

That's why I think it's more consistent and reduces the potential for confusion to not support that alias in the context of RG. It indicates it's a special case.

Reply:
as someone who used these BIT instructions for their project, I found it confusing to use R3/RS in place of what I was really doing, trying to read from the IN port. It seems more ergonomic to me to use something like IN/OUT here to say what I'm really doing.

I do understand your argument as well, though. Would a different, similar alias be more to your liking, while still being able to visually differentiate between IN vs OUT, which RS does not? `RS_IN`, `R_IN`, `PORT_IN`?

Reply:
My main argument is really in the interest of portability. Whatever code is documented in Voja's PDFs has a high chance of working across different implementations and adding additional functionality will force future assemblers to have to implement all these behaviors for compatibility with existing code. I do recognize that these arguments are a bit idealistic given the small scale of the project so far :)

I think a library of aliases (i.e. EQUs) to add some additional convenience definitions may be more appropriate.

I personally would prefer RS and try to avoid R3 (although that wasn't an option before this change). In general assembly code does require the reader to be familiar with the low level details of the language and not knowing what RS means a reader would have to look it up, but then would see the special behavior that is associated with that.

Reply:
I had the same thought, re: using EQU instead for aliasing. I've been contemplating tackling #include directive so we could throw all these useful aliases into a badge.inc file that could be distributed with the assembler.

In the meantime, I'll remove the IN/OUT aliases from the PR, however much it pains me to do so.